kinto_megaphone/listeners.py

```diff
@@ -0,0 +1,50 @@
+from kinto.core.events import ResourceChanged
+from kinto.core.listeners import ListenerBase
+from kinto.core import utils
+from . import megaphone
+
+DEFAULT_SETTINGS = {}
+
+
+class CollectionTimestamp(ListenerBase):
+    """An event listener that pushes all collection timestamps to Megaphone."""
+    def __init__(self, client, broadcaster_id):
+        self.client = client
+        self.broadcaster_id = broadcaster_id
+
+    def __call__(self, event):
+        if not isinstance(event, ResourceChanged):
+            return
```

Comment:
I'm guessing you're asking because you'd prefer doing it in config. I'm of two minds about this. On the one hand, it's possible to do it in config already so why add an extra mechanism? But on the other hand, there's really only one correct way to use this listener -- to keep Megaphone versions in sync with collection ETags. Since there's only one correct way, I figured it made sense to formalize this way in code. There may be more flexible uses of this code that I didn't anticipate but I decided to just do the simple thing because of YAGNI.

I tried to explain this in the PR notes, but maybe I didn't do a great job.

Reply:
OK I see... yeah let's keep it super explicit like this

Reply:
It turns out that unless the configuration of the listener is explicitly specified, the listener is only subscribed to "write" events, so this probably isn't actually needed. See https://github.com/Kinto/kinto/blob/1af1f00e7e106b741ca8e94d7fd6de98e1442a47/kinto/core/initialization.py#L454-L458.